hydra_agent/agent.py

```diff
@@ -0,0 +1,32 @@
+from hydra_agent.requests import Requests
+from hydra_agent.redis_proxy import RedisProxy
+from hydra_agent.sync_mechanism import SynchronizationProcessing
+from requests import Session
+
+
+class HydraAgent(Session):
+    def __init__(self, entrypoint_url):
+        self.entrypoint_url = entrypoint_url
+        self.redis_proxy = RedisProxy()
+        self.sync_processing = SynchronizationProcessing(entrypoint_url,
```

Comment:
Yea it is fine. Instead of using inheritance we use composition/wrapping of an instance from another class

Reply:
I would prefer a function for every verb: `self._sync_get(...)` so it is more controllable and we avoid having `if...else` blocks for verbs.

Reply:
There is a function for every verb at [sync_mechanism.py ](https://github.com/HTTP-APIs/hydra-python-agent/pull/109/files#diff-22bb81c6fd017d49156ccfa7a9a90046)(sync_get(...), sync_delete(...))...  add_operation basically calls the right function.

I added this layer between(add_operation) because that's the way the information is going to be received from the Modification Table probably: | METHOD | RESOURCE_url | ...